src/pages/search/models.tsx

```diff
@@ -0,0 +1,156 @@
+import { useInView } from 'react-intersection-observer';
+import { useEffect } from 'react';
+import { ModelGetAll } from '~/types/router';
+import { ModelCard } from '~/components/Cards/ModelCard';
+import { SearchHeader } from '~/components/Search/SearchHeader';
+
+const searchClient = instantMeiliSearch(
+  env.NEXT_PUBLIC_SEARCH_HOST as string,
+  env.NEXT_PUBLIC_SEARCH_CLIENT_KEY,
+  { primaryKey: 'id', keepZeroFacets: true }
+);
+
+export default function Search() {
+  return (
+    <InstantSearch searchClient={searchClient} indexName="models" routing={routing}>
+      <Container fluid>
+        <Stack
+          sx={(theme) => ({
+            height: 'calc(100vh - 2 * var(--mantine-header-height,50px))',
+            position: 'fixed',
+            left: 0,
+            top: 'var(--mantine-header-height,50px)',
+            width: '377px',
+            overflowY: 'auto',
+            padding: theme.spacing.md,
+          })}
+        >
+          <RenderFilters />
+        </Stack>
+
+        <Stack pl={377} w="100%">
+          <SearchHeader />
+          <ModelsHitList />
+        </Stack>
+      </Container>
+    </InstantSearch>
+  );
+}
+
+const RenderFilters = () => {
+  return (
+    <>
+      <SortBy
+        title="Sort models by"
+        items={[
+          { label: 'Highest Rated', value: 'models:metrics.weightedRating:desc' },
+          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
+          { label: 'Most Liked', value: 'models:metrics.favoriteCount:desc' },
+          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
+          { label: 'Newest', value: 'models:createdAt:desc' },
+        ]}
+      />
+      <ChipRefinementList
+        title="Filter by Base Model"
+        attribute="modelVersion.baseModel"
+        sortBy={['name']}
+      />
+      <ChipRefinementList title="Filter by Model Type" attribute="type" sortBy={['name']} />
+      <ChipRefinementList
+        title="Filter by Checkpoint Type"
+        sortBy={['name']}
+        attribute="checkpointType"
+      />
+      <SearchableMultiSelectRefinementList
+        title="Tags"
+        attribute="tags"
+        // TODO.Search: Meiliserach & facet searching is not supported when used with sort by as Angolia provides it.  https://github.com/meilisearch/meilisearch-js-plugins/issues/1222
+        // If that ever gets fixed, just make sortable true + limit 20 or something
+        limit={9999}
+        searchable={false}
+      />
+    </>
+  );
+};
+
+const useStyles = createStyles((theme) => ({
+  grid: {
+    display: 'grid',
+    gridTemplateColumns: `repeat(auto-fill, minmax(250px, 1fr))`,
+    columnGap: theme.spacing.md,
+    gridTemplateRows: `auto 1fr`,
+    overflow: 'hidden',
+    marginTop: -theme.spacing.md,
+
+    '& > *': {
+      marginTop: theme.spacing.md,
+    },
+  },
+}));
+
+export function ModelsHitList() {
+  const { hits, showMore, isLastPage } = useInfiniteHits();
```

Comment:
We now have a type coming from the search index 🙂 just make sure to get the import
```suggestion
  const { hits, showMore, isLastPage } = useInfiniteHits<ModelSearchIndexRecord>();
```